### DISCUSSION TITLE
Github contribution is not up

### DISCUSSION THREAD

**Discussion by hassamarifdev** [community/community#141870] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

my github contribution is not updated its been several weeks I guess.. just checked and what I saw shocked me that my github contribution is stuck for unknown reason. 

as you can see in the image 
<img width="547" alt="image" src="https://github.com/user-attachments/assets/de8ade5d-8b37-4c14-9379-acae63f76a5f">

according to Activity review in the bottom of the image I updated that report 11 minutes ago but can't see any change in my contribution card.

**Comment by melvincwng** [comment]
@hassamarifdev, it could be because your contributions are in *private repositories* which are not shown on the Contribution Activity by default.

Consider changing the visibility of your private contributions by following this documentation link [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/showing-your-private-contributions-and-achievements-on-your-profile#changing-the-visibility-of-your-private-contributions)

Hope this answers your question 😄

**Reply by hassamarifdev** [reply]
yep, but visibility for private contributions is on. still, their contribution card is not updated...

**Reply by goodLikes** [reply]
This can be caused by several reasons:

Caching: The browser or GitHub can cache an old version of your profile page.
Network problems: An unstable connection may prevent data from being updated.
Errors in the GitHub configuration: Incorrect repository or account settings may affect the display of contributions.
Update delay: Sometimes updating contributions may take some time.

Possible solutions:

Clear the browser cache:

Completely clear the cache and cookies: This will help to reload the page from scratch.
Try a different browser: Sometimes the problem may be related to a specific browser.


Check your privacy settings:

Make sure that you don't have browser extensions blocking GitHub content.


Disable the GitHub cache:
Use secret mode: Open GitHub in incognito mode or private mode.

Check the repository settings:

Make sure that the repository is public: Private repositories may not be displayed in the profile.
Check the visibility settings: Make sure that your commits are not hidden.

Check your profile settings:

Make sure that your GitHub email address is linked to your profile: This is necessary for the correct tracking of deposits.

Wait for a while:
Sometimes it can take several hours to update deposits.


Contact GitHub support:
If none of the above methods helped, contact GitHub support. They will be able to check your account for any problems.


Additional tips:

Check the GitHub documentation: In the GitHub documentation, you will find more detailed information about profile settings and the display of contributions.
Look for information on the GitHub forums: Perhaps someone has already faced a similar problem and found a solution.
Check the GitHub status: Sometimes problems with displaying contributions may be due to temporary failures on GitHub servers.

Have you tried to fork the repository and send a pull request there? It may help to check if your changes are being displayed.
Have you used any tools to automate deposits? Some tools may affect the display of contributions.
Have you recently changed your nickname on GitHub? Changing the nickname may cause temporary problems with displaying contributions.

**Reply by hassamarifdev** [reply]
nope, that is not the problem at all. yes repository is private but I'm I allowed to show private contributions.

**Comment by Dhruvnet** [comment]
Hey there,

I also faced a similar issue with my GitHub contributions. The problem stemmed from a bot I was using that indicated my commits were successfully made in the repository, but in reality, those commits never occurred.

To address this issue, here’s how I attempted to resolve it:

1. Review Contribution History
I first examined my profile to identify any discrepancies in my contribution history. It turned out that contributions were recorded for a specific period, but later contributions seemed to vanish.

2. Verify Commit Author Information
Next, I ensured that my commits were associated with the correct email address. I ran the following command in my terminal to check my current Git configuration:
git config --global user.email
If the email did not match my GitHub account, I updated it using:
git config --global user.email "your_email@example.com"
This helped me confirm that my contributions would be correctly attributed to my profile.

3. Test with a Public Repository
I then changed the visibility of my repository to public temporarily. I made a new commit manually, ensuring it was not from any automated tool or bot. This allowed me to see if the contributions registered correctly in my contribution graph.

By following these steps, I was able to pinpoint the issue and ensure that my contributions were accurately reflected on my GitHub profile. I hope this helps you resolve your situation as well!

**Reply by hassamarifdev** [reply]
> Hey there,
> 
> I also faced a similar issue with my GitHub contributions. The problem stemmed from a bot I was using that indicated my commits were successfully made in the repository, but in reality, those commits never occurred.
> 
> To address this issue, here’s how I attempted to resolve it:
> 
> 1. Review Contribution History
>    I first examined my profile to identify any discrepancies in my contribution history. It turned out that contributions were recorded for a specific period, but later contributions seemed to vanish.
> 2. Verify Commit Author Information
>    Next, I ensured that my commits were associated with the correct email address. I ran the following command in my terminal to check my current Git configuration:
>    git config --global user.email
>    If the email did not match my GitHub account, I updated it using:
>    git config --global user.email "[your_email@example.com](mailto:your_email@example.com)"
>    This helped me confirm that my contributions would be correctly attributed to my profile.
> 3. Test with a Public Repository
>    I then changed the visibility of my repository to public temporarily. I made a new commit manually, ensuring it was not from any automated tool or bot. This allowed me to see if the contributions registered correctly in my contribution graph.
> 
> By following these steps, I was able to pinpoint the issue and ensure that my contributions were accurately reflected on my GitHub profile. I hope this helps you resolve your situation as well!

I checked my email before raising this issue, and it is correct. When tested with a public repository, it works fine, but not with the private project I'm working on. One thing I noticed is that it started happening when I created another branch for the backend and started pushing my code there. Is there anything that you can guide me with? [Dhruvnet](https://github.com/Dhruvnet)

**Reply by Dhruvnet** [reply]
Heyy 
If your email works fine with public repos but not with a private project, and issues begin after creating a new branch, it could be:
1. Access Issues: Ensure you have the right permissions for the private repo and branch.

2. SSH vs. HTTPS: Verify if your repo uses SSH or HTTPS, and check if your authentication (like an SSH key or personal access token) is configured properly.

3. Remote URL: Check if the remote URL for your repo is correct with git remote -v, and update if needed (git remote set-url origin <new-url>).

4. Authentication Tokens: Sometimes, access tokens expire or need updating, especially if your backend settings or branch permissions changed recently.


Most likely it seems to be an Access issue or auth token !! 
If this didn't work can you specify more on what logic you build in the backend ?

**Comment by YasinDehfuli** [comment, marked as answer]
Hey, dear @hassamarifdev! 

You should contact the GitHub support team: https://support.github.com/.

Go to My Tickets & create new ticket

Admins will help you to resolve the problem.

**Reply by hassamarifdev** [reply]
I will. thank you 👍
